app/models/hubstats/label.rb

```diff
@@ -1,13 +1,15 @@
 module Hubstats
   class Label < ActiveRecord::Base
-    scope :with_a_pull_request, lambda { |pull_ids|
-      select("hubstats_labels.*")
-      .select("COUNT(hubstats_labels_pull_requests.pull_request_id) AS pull_request_count")
-      .joins("LEFT JOIN hubstats_labels_pull_requests ON hubstats_labels_pull_requests.label_id = hubstats_labels.id")
-      .with_ids(pull_ids)
-      .having("pull_request_count > 0")
-      .group("hubstats_labels.id")
-    }
+
+  scope :with_a_pull_request, lambda { |params|
+    select("hubstats_labels.*")
+    .select("COUNT(hubstats_labels_pull_requests.pull_request_id) AS pull_request_count")
+    .joins(:pull_requests).merge(Hubstats::PullRequest.with_state(params[:state])
+                                                      .belonging_to_users(params[:users])
+                                                      .belonging_to_repos(params[:repos]))
+    .having("pull_request_count > 0")
+    .group("hubstats_labels.id")
+  }
```

Comment:
So, I know this was the basic suggestion that I provided, but I think it is a bit unclear and dense.  Consider breaking it up and renaming it to something like this:

```ruby
def index
  pull_requests = PullRequest.all_filtered(params)
  @labels = Hubstats::Label.count_by_pull_request(pull_requests)
  
  # ...
end

```ruby
# class Hubstats::Label

def self.counts_by_pull_request(pull_requests)
  select("hubstats_labels.*")
    .select("COUNT(hubstats_labels_pull_requests.pull_request_id) AS pull_request_count")
    .joins(:pull_requests).merge(pull_requests)
    .having("pull_request_count > 0")
    .group("hubstats_labels.id")
    .order("pull_request_count DESC")
end
```

```ruby
# class Hubstats::PullRequest

def self.all_filterd(params)
  with_state(params[:state])
    .belonging_to_users(params[:users])
    .belonging_to_repos(params[:repos]))
end
```

You might even be able to break what I have in the `::counts_by_pull_request` method up a bit more if you wanted.